main.go

```diff
@@ -1548,6 +1548,11 @@ func main() {
 		options.PrintCommands = printCommand
 	}
 
+	// Compatibility with legacy -target=wasi
+	if options.Target == "wasi" {
+		options.Target = "wasip1"
+	}
```

Comment:
I'd prefer if you added a file targets/wasi.json with just the following contents:

```json
{
    "inherits": ["wasip1"]
}
```

This avoids special-casing WASI in the compiler and this trick is already used for targets/clue.json for example.